admin-notice/class-admin-notice.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace Automattic\VIP\Admin_Notice;
+
+class Admin_Notice {
+	public $message;
+	public $start_date;
+	public $end_date;
+
+	/**
+	 * Create AdminNotice
+	 *
+	 * @param {string} $message - the text to be displayed
+	 * @param {string} $start_date - the date* since the notice should start to show up
+	 * @param {string} $end_date - the date* till the notice should show up
+	 *
+	 * * Date Format should follow 'Day-Month-Year Hour:Minute' format.
+	 */
+	public function __construct( string $message, string $start_date, string $end_date ) {
```

Comment:
Are there cases where it wouldn't make sense to constrain a notice by time? Think about the notice we show when an account billing is not up do date - we don't know what the end time should be, so while one could enter a date with a far-distant year, that's really a hack of poor class design.

I'm thinking whether it would not be overkill to have Admin_Notice just take in a message and a notice type (make them class constants somewhere) as args, and then have a `Time_Constrained_Notice()` decorator that handled that, along with my previous `Editor_Notice()` (I guess `Role_Constrained_Notice()` or `Capability_Constrained_Notice()`) constraint suggestion.

```php
// Show to everyone, but only for a limited time.

$admin_notice_controller->add(
	new Time_Constrained_Notice(
		new Admin_Notice( 'Happy New Year!', Admin_Notice_Type::INFO ),
		'01-01-2021 00:00',
		'02-01-2021 00:00'
	)
);

// Plugin updates from a different PR. Wrap this in it's own logic of when to show.

$admin_notice_controller->add(
	new Capability_Constrained_Notice(
		new Admin_Notice( 'Plugin updates should be done through GitHub', Admin_Notice_Type::Warning ),
		'update_plugins'
	)
);

// Lockout Error (example).

$admin_notice_controller->add(
	new Capability_Constrained_Notice(
		new Admin_Notice( 'You have not paid your bill.', Admin_Notice_Type::Error ),
		'manage_options'
	)
);
```